common/djangoapps/dark_lang/api.py

```diff
@@ -0,0 +1,199 @@
+"""
+Support code for dark-launching languages.  These languages won't be used
+when determining which translation to give a user based on their browser
```

Comment:
To me, it doesn't feel like this code belongs in an "api" file. Api files are usually used for internal apis that will be called from a number of places. Typically a request object is not passed into the methods.

Why did you decide to move this code out of middleware.py? I don't find that it helps with readability, as both files touch preferences, request objects, etc.